app/routers/profile.py

```diff
@@ -148,3 +144,80 @@ def get_image_crop_area(width, height):
         return (delta, 0, width - delta, height)
     delta = (height - width) // 2
     return (0, delta, width, width + delta)
+
+
+@router.get("/import_holidays")
+def import_holidays(request: Request):
+    # Made up user details until there's a user login system
+    current_username = "Chuck Norris"
+
+    return templates.TemplateResponse("holidays.html", {
+        "request": request,
+        "username": current_username
+    })
+
+
+@router.post("/update_holidays")
+async def update_holidays(
+        file: UploadFile = File(...), session=Depends(get_db)):
+    icsfile = await file.read()
+    holidays = get_holidays_from_file(icsfile.decode(), session)
+    try:
+        await save_holidays_to_db(holidays, session)
+    except Exception as ex:
+        print(ex)
+    finally:
+        url = router.url_path_for("profile")
+        return RedirectResponse(url=url, status_code=HTTP_302_FOUND)
+
+
+def get_holidays_from_file(file, session):
+    """
+    this function using regex to extract holiday title
+    and date from standrd ics file
+    :param file:standard ics file
+    :param session:current connection
+    :return:list of holidays events
+    """
+    regex = re.compile(
```

Comment:
Prefer compiling it outside the function to prevent compilation every time the function is called

Reply:
i read in [SO](https://stackoverflow.com/questions/3427329/compiling-a-regex-inside-a-function-thats-called-multiple-times) that re.compile use cache so im not sure about the difference - anyway, ill move it to global